### PR TITLE
API refactor

### PR DESCRIPTION
**Pull request by bnaul**
- Store feature info in a `pd.DataFrame` instead of `xr.Dataset`; for multi-channel features, columns labels are a `(feature, channel)` tuple.
- Since feature data is simply rectangular now, there's no need for the `build_model`/`predict` wrapper functionality.
- Otherwise haven't changed the `featurize` API yet.

Some open questions:
- What functionality would be useful to add on top of `sklearn` for model building/prediction?
- What part of the `Featureset` class do we still want to include?
    - Is `.impute` redundant w/ `sklearn`'s imputation functionality? 
    - Definitely want some form of serialization besides CSV; what about `joblib` instead of `netCDF4`? We already use it for models on the front end.

### CODE REVIEW

**Comment by stefanv** [comment]
As far as I know, joblib uses pickles to serialize.  This is not transportable.  We could use npy or npz, with a thin wrapper for handling unpacking back into a pandas array.

**Comment by stefanv** [comment]
Would it make sense to provide a utility function to extract all channels of a certain type from a DataFrame?

**Comment by acrellin** [comment]
Yes, joblib does use pickle under the hood.

**Comment by bnaul** [comment]
What would the wrapper entail? .npy works for the actual data but then do we need a separate file that stores the metadata (row/column names)?

**Comment by bnaul** [comment]
@stefanv re: extracting different channels: yeah, this is hard if we use tuples as column names, i.e.
```
       (mean, 0)  (mean, 1)     meta1     meta2
file0        2.0   0.180734  0.548427  0.187956
file1        2.5   0.196072  1.171789  0.174803
file2        3.0   0.558093  0.265003  0.109842
file3        2.6   0.280734  0.448427  0.287956
file4        2.5   0.496072  1.371789  0.274803
file5        3.0   0.258093  0.465003  0.409842
```
If we use a multi-index instead this is easy:
```
feature mean               meta1     meta2
channel    0         1
file0    2.0  0.180734  0.548427  0.187956
file1    2.5  0.196072  1.171789  0.174803
file2    3.0  0.558093  0.265003  0.109842
file3    2.6  0.280734  0.448427  0.287956
file4    2.5  0.496072  1.371789  0.274803
file5    3.0  0.258093  0.465003  0.409842
```
but then the columns that don't have a `channel` associated with them are harder to access than they should be. I'm on the fence about this, what do y'all think?

**Comment by stefanv** [comment]
npz can store multiple arrays in one file (i.e., list of column names + data)

It may be preferable to have a consistent format for accessing columns with/without channel data, i.e. pretend that single channel is multi-channel with index 0

**Comment by bnaul** [comment]
Hm npz would be nice, maybe I'll write a little wrapper around that.

I can see both (multiindex or tuples) being annoying, perhaps `featurize` can have a flag that chooses which format is used? It's also not hard for the user to make the conversion manually to whatever they prefer but supporting both formats explicitly might be nicer.

**Comment by stefanv** [comment]
Probably best for us to choose one way to do it (even if sub-optimal in all use cases), otherwise confusion is bound to arise.

**Comment by stefanv** [comment]
@bnaul I now configured the pep complainer to use 80, not 79 :)

File: cesium/featurize.py [hunk 1]
```diff
@@ -64,34 +62,27 @@ def featurize_single_ts(ts, features_to_use, custom_script_path=None,
...
-    an xarray.Dataset.
+    a pd.DataFrame.
 
     Parameters
     ----------
-    feature_dicts : list of dict
-        List of dicts (one per time series file) with feature names as keys and
-        lists of feature values (one per channel) as values.
+    features_list : list of pd.Series
+        List of seires (one per time series file) with (feature name, channel)
```
Comment (stefanv) [inline cesium/featurize.py]: series

File: cesium/featurize.py [hunk 2]
```diff
@@ -329,19 +284,81 @@ def featurize_ts_files(ts_paths, features_to_use, output_path=None,
...
+def impute_featureset(fset, strategy='constant', value=None, max_value=1e32,
+                      inplace=False):
+    """Replace NaN/Inf values with imputed values as defined by `strategy`.
+    Output should satisfy `sklearn.validation.assert_all_finite` so that
+    training a model will not produce an error.
+
+    Parameters
+    ----------
+    strategy : str, optional
+    The imputation strategy. Defaults to 'constant'.
```
Comment (stefanv) [inline cesium/featurize.py]: Docstring indentation

File: cesium/featurize.py [hunk 3]
```diff
@@ -329,19 +284,81 @@ def featurize_ts_files(ts_paths, features_to_use, output_path=None,
...
+    good choice for e.g. random forests).
+
+    max_value : float, optional
+    Maximum (absolute) value above which values are treated as infinite. Used
+    to prevent overflow when fitting `sklearn` models.
+
+    Returns
+    -------
+    pd.DataFrame
+    Feature data frame wth no missing/infinite values.
```
Comment (stefanv) [inline cesium/featurize.py]: Indentation

**Comment by bnaul** [comment]
All green (except `cesium_web`)! Ready for y'all to take a proper look if you like 🤓

**Comment by stefanv** [comment]
This looks great to me!  But: please note that no-one has yet fixed the dependent build server, so you definitely cannot trust that result :)

**Comment by acrellin** [comment]
Great how much simpler this makes things! :+1: from me.
